crates/bevy_debug_draw/src/lib.rs

```diff
@@ -0,0 +1,155 @@
+use bevy_app::{CoreStage, Plugin};
+use bevy_asset::{load_internal_asset, Assets, HandleUntyped};
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    query::With,
+    system::{Commands, Query, Res, ResMut, Resource},
+};
+use bevy_reflect::TypeUuid;
+use bevy_render::{
+    prelude::{Mesh, SpatialBundle},
+    render_phase::AddRenderCommand,
+    render_resource::{PrimitiveTopology, Shader, SpecializedMeshPipelines},
+    Extract, RenderApp, RenderStage,
+};
+
+#[cfg(feature = "bevy_pbr")]
+use bevy_pbr::{NotShadowCaster, NotShadowReceiver};
+#[cfg(feature = "bevy_sprite")]
+use bevy_sprite::Mesh2dHandle;
+
+pub mod debug_draw;
+
+#[cfg(feature = "bevy_sprite")]
+pub mod pipeline_2d;
+#[cfg(feature = "bevy_pbr")]
+pub mod pipeline_3d;
+
+use crate::debug_draw::DebugDraw;
+
+/// The `bevy_debug_draw` prelude.
+pub mod prelude {
+    #[doc(hidden)]
+    pub use crate::{debug_draw::DebugDraw, DebugDrawConfig, DebugDrawPlugin};
+}
+
+pub const SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 7414812689238026784);
+
+pub struct DebugDrawPlugin;
+
+impl Plugin for DebugDrawPlugin {
+    fn build(&self, app: &mut bevy_app::App) {
+        load_internal_asset!(app, SHADER_HANDLE, "debuglines.wgsl", Shader::from_wgsl);
+
+        app.init_resource::<DebugDraw>()
+            .init_resource::<DebugDrawConfig>()
+            .add_system_to_stage(CoreStage::Last, update);
+
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else { return; };
+
+        render_app.add_system_to_stage(RenderStage::Extract, extract);
+
+        #[cfg(feature = "bevy_sprite")]
+        {
+            use bevy_core_pipeline::core_2d::Transparent2d;
+            use pipeline_2d::*;
+
+            render_app
+                .add_render_command::<Transparent2d, DrawDebugLines>()
+                .init_resource::<DebugLinePipeline>()
+                .init_resource::<SpecializedMeshPipelines<DebugLinePipeline>>()
+                .add_system_to_stage(RenderStage::Queue, queue);
+        }
+
+        #[cfg(feature = "bevy_pbr")]
+        {
+            use bevy_core_pipeline::core_3d::Opaque3d;
+            use pipeline_3d::*;
+
+            render_app
+                .add_render_command::<Opaque3d, DrawDebugLines>()
+                .init_resource::<DebugLinePipeline>()
+                .init_resource::<SpecializedMeshPipelines<DebugLinePipeline>>()
+                .add_system_to_stage(RenderStage::Queue, queue);
+        }
+    }
+}
+
+#[derive(Resource, Clone, Copy)]
+pub struct DebugDrawConfig {
```

Comment:
Could this be merged into `DebugDraw` itself?

Reply:
This was split because it's awkward to have private fields on a config struct like this. It also gets sent to the render world.